public/app/core/services/NewFrontendAssetsChecker.ts

```diff
@@ -1,41 +1,48 @@
-import { isEqual } from 'lodash';
-
 import { getBackendSrv, getGrafanaLiveSrv } from '@grafana/runtime';
 
-import { config } from '../config';
-
 export class NewFrontendAssetsChecker {
-  private _hasUpdates = false;
+  private hasUpdates = false;
+  private previous = '?';
+  private checked = 0;
+  private interval = 1000; // check interval
 
   public start() {
     let prevState: boolean | undefined;
 
-    // Subscribe to live connecttion state changes and check for new assets when re-connected
-    getGrafanaLiveSrv()
-      .getConnectionState()
-      .subscribe((connected) => {
+    // Subscribe to live connection state changes and check for new assets when re-connected
+    const live = getGrafanaLiveSrv();
+    if (live) {
+      live.getConnectionState().subscribe((connected) => {
         if (connected && connected !== prevState) {
           this._checkForUpdates();
         }
       });
+    }
   }
 
   private async _checkForUpdates() {
-    if (this._hasUpdates) {
+    if (this.hasUpdates) {
       return;
     }
 
-    const result = await getBackendSrv().get('/api/frontend/assets');
-
-    if (!isEqual(result, config.bootData.assets)) {
-      this._hasUpdates = true;
+    const resultRaw = await getBackendSrv().get('/api/frontend/assets');
+    const result = JSON.stringify(resultRaw);
+    if (this.checked && this.previous !== result) {
+      this.hasUpdates = true;
       console.log('updates detected', true);
     }
+    this.previous = result;
+    this.checked = Date.now();
   }
 
   public reloadIfUpdateDetected() {
-    if (this._hasUpdates) {
+    if (this.hasUpdates) {
       window.location.reload();
```

Comment:
didn't we want to do it during navigation? Otherwise it will just refresh in the middle of someone editing a dashboard.

Reply:
@tolzhabayev this function is only called when moving to another page

Reply:
got it, thanks for clarifying.